### PR TITLE
[nats] Fixed Alpine Dockerfile that was not using config file

### PR DESCRIPTION
**Pull request by kozlovic**
Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

**Comment by kozlovic** [comment]
@tianon @yosifkit We had the Alpine Dockerfile not specifying entrypoint and using config file (see https://github.com/nats-io/nats-docker/commit/ea417afe07d36ec3c0b20032c395bc9e54dedd14).
I am wondering if we update the official image or not (hence why this PR is in draft). Say that we do and I have single commit at the top of the library file. Will that regenerate all images or simply the alpine? If if updates all, would that be better that I am git commit for each image and change only for the alpine one? Thanks!

**Comment by kozlovic** [comment]
oh.. but then now the image fails with the override-cmd, so I would need to change test/config.sh to exclude with what, that `[nats:alpine_override-cmd]=1`?

**Comment by yosifkit** [comment]
No, the `override-cmd` test should not be disabled.  Since the image has a shell and other commands that can be used to explore the container, it needs to follow the "consistency" guideline: 

> A beginning user should be able to `docker run official-image bash` (or `sh`) without needing to learn about `--entrypoint`
>
> \- https://github.com/docker-library/official-images/tree/c39fe063841e2fecbb89c8125bbdaeff86730cd3#consistency

(there is an example there of providing an entrypoint script so that `--flag` as well as `sh` work)

**Comment by yosifkit** [comment]
To answer the building question: the build servers make use of docker build cache so since the `windowsservercore` images have unchanged Dockerfiles they won't be pushed again.

**Comment by kozlovic** [comment]
@yosifkit Thanks a bunch! I think I understand what I have to do. I am going to update/test the nats-docker repo and will update the PR here to see if that fixes this.

**Comment by kozlovic** [comment]
@yosifkit Ok, I think I got it. Thank you for your continued patience with me :-)

**Comment by yosifkit** [comment]
Did you want your entrypoint to have automatic `--config /etc/nats/nats-server.conf` added even if users add their own flags, like: `docker run -d nats:alpine -D`? I can go ahead and merge as-is if you need to think on it.

**Comment by kozlovic** [comment]
> Did you want your entrypoint to have automatic --config /etc/nats/nats-server.conf added even if users add their own flags, like: docker run -d nats:alpine -D?

Are you saying that in this case it would automatically use config AND the flags? If so, I don't think so because then there would be no way of NOT using what is in the config file I would think? Meaning that one would have to override with flag all settings in the config file that one does not want. If one day we extend this config, it may not even be possible since not everything can be set from the command line. But maybe I am not understanding what you were saying?

Basically the goal is that if one runs `docker run nats:alpine`, it uses the default configuration. After that, if one provides any flag, then the configuration is ignored and the server is configured from the command line flags. Of course, one of the flag can be `-c <config file>` and combine flags.

**Comment by kozlovic** [comment]
Also, with current approach one can always do `docker run nats:alpine -c /etc/nats/nats-server.conf -D` which would combine provided config file + debug flag. But again, at least we are not forcing the config file.

**Comment by kozlovic** [comment]
> This section is not strictly necessary (since exec "$@" is what follows), but is harmless.

I see, in the example it was there because of using `gosu`.. I may update that for the next NATS Server release, but ok for now
